File: PA6.cpp


#include<iostream>
using namespace std;

class ArrayX
{
    private:
        int *Arr;
        int iSize;
		int iNo;

    public:
	    int iCnt = 0;
		void Accept()       
        {
            cout<<"Enter the value of constructor : "<<endl;
            cin>>this->iNo;
        }
        
        ArrayX(int iValue)               
        {
            this->iSize = iValue;
            Arr = new int[iSize];       
        }

        ~ArrayX()                       
        {
            delete []Arr;               
        }
        
	    
		void Accept1()   
        {
            int iCnt = 0;
            cout<<"Enter the elements"<<endl;
            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                    cin>>Arr[iCnt];
            }
        }
		
		int Minimum()
        {
            int iCnt = 0;
			int iMin = Arr[0];

            cout<<"Elements of array are "<<endl;
            for(iCnt = 0; iCnt< iSize; iCnt++)
            {
                if(iMin > Arr[iCnt])
				{
					iMin = Arr[iCnt];
				}
            }
			return iMin;
        }

       

      
};

int main()
{
    ArrayX obj1(5);
	int iRet = 0;

    obj1.Accept();           
    obj1.Accept1();   
    
    iRet = obj1.Minimum();
    cout<<"Minimum number is: "<<iRet<<endl;	
             

 

    return 0;
}

File: PA6.java
// Template for probplems on N Numbers

import java.lang.*;
import java.util.*;

class ArrayX
{
    private int Arr[];
	private int iNo = 0;

    public ArrayX(int iNo)
    {
        Arr = new int[iNo];
    }

    public void Accept()
    {
        int iCnt = 0;

        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the values : ");

        for(iCnt = 0; iCnt < iNo; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public void Accept1()
    {
        int iCnt = 0;

        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the values to search : ");

        iNo = nextInt();
    }

    public boolean Search()
	{
		int iCnt = 0;
		boolean bFlag = false;
		
		for(iCnt = 0; iCnt < Arr.length;iCnt++)
		{
			if(Arr[iCnt] == iNo)
			{
				bFlag = true;
				break;
			}
		}
		return bFlag;
	}
}

class PA6
{
    public static void main(String arg[])
    {
            Scanner sobj = new Scanner(System.in);
            int iLength = 0;
			boolean iRet;

            System.out.println("Enter the value for constructor");
            iLength = sobj.nextInt();
		

            ArrayX obj = new ArrayX(iLength);

            obj.Accept();
         
			
			
			obj.Accept1();

            iRet = obj.Search();
			
			if(iRet == true)
			{
				System.out.println("Element is there in array");
			}
			else 
			{
				System.out.println("Element is not there");
			}

            obj = null;
    }
}